SortingAlgorithms/Algorithms/Searching/BinarySearch.cs

```diff
@@ -9,46 +9,60 @@
 
 namespace SortingAlgorithms.Algorithms.Searching
 {
-    public class BinarySearch : ISearchingAlgorithm
+    public sealed class BinarySearch : ISearchingAlgorithm 
     {
-        public void Search(int[] array, int number)
+        private readonly ISortingAlgorithm _sortingAlgorithm;
+        private readonly dynamic _arrayValidator;
```

Comment:
Dynamic is a bad practice, for me.
Create interface IArrayValidator and use the same way as ISortingAlgorithm